nemo/collections/nlp/data/text_classification/text_classification_dataset.py

```diff
@@ -60,193 +60,215 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
             'input_ids': NeuralType(('B', 'T'), ChannelType()),
             'segment_ids': NeuralType(('B', 'T'), ChannelType()),
             'input_mask': NeuralType(('B', 'T'), MaskType()),
-            'label': NeuralType(('B'), LabelsType()),
+            'label': NeuralType(('B',), LabelsType()),
         }
 
     def __init__(
         self,
-        input_file: str,
-        tokenizer: Any,
-        max_seq_length: int,
+        tokenizer: TokenizerSpec,
+        input_file: str = None,
+        queries: List[str] = None,
```

Comment:
Ok, in this solution you added other "source of data": input_file vs queries.

If we will follow this then all the datasets used in the TLT should have this "other source of data" added to. That will complicate the dataset's logic and make them even more complex. Let's discuss pros and cons.

Reply:
It is for unifying the place to perform the processing data, one of the things we discussed in the last meeting. Lets discuss it.